src/RepoToolset/ProjectDefaults.props

```diff
@@ -61,18 +68,19 @@
       </Choose>
     </When>
   </Choose>
-  
+
   <Import Project="|Build precondition failed: Unable to determine the language of the project. Consider adding Directory.Build.props that sets Language property to directory '$(MSBuildProjectDirectory)'.|"
           Condition="'$(Language)' == '' and '$(MSBuildProjectExtension)' != '.shproj'"/>
-  
+
   <Choose>
     <!-- C# specific settings -->
     <When Condition="'$(Language)' == 'C#'">
       <PropertyGroup>
         <CheckForOverflowUnderflow>true</CheckForOverflowUnderflow>
-        <NoWarn>$(NoWarn);1591</NoWarn>
+        <NoWarn Condition="'$(NoWarn)' == ''">1701;1702;1705;1591</NoWarn>
+        <ProjectLanguage>CSharp</ProjectLanguage>
```

Comment:
><ProjectLanguage>CSharp</ProjectLanguage> [](start = 8, length = 41)

What is this property used for?

Reply:
SDK/CLI use it in a number of places. Someone who has been on the team longer may be able to comment if it is actually important.

Reply:
Perhaps the repo should use ```$(Language)``` instead, or define ProjectLanguage in it's private props files if that makes things cleaner for some reason. I would rather avoid polluting all repos with yet another property that specifies the language.